source/fundamentals/object-serialization.txt

```diff
@@ -0,0 +1,241 @@
+data between Extended JSON and Java objects. You can specify how the
+driver should perform serialization using **codecs**.
+
+Codecs implement translation between Extended JSON and Java objects
+using methods that encode and decode data.
+Each ``CodecRegistry`` contains a list of codecs. When you specify a 
+``CodecRegistry`` for a ``MongoClient``, ``MongoDatabase``, or
+``MongoCollection``, the ``CodecRegistry`` is used for all subsequent
+operations on that object. Whenever you send or receive data from a
+connected MongoDB instance, the driver selects the first codec that can
+translate each object in the ``CodecRegistry`` of the
+``MongoClient``, ``MongoDatabase``, or ``MongoCollection`` used to
+communicate with the MongoDB instance.
+
+.. ref:: default-codecs
+
+Default Codecs
+--------------
+
+By default, every ``MongoClient`` uses a ``CodecRegistry`` that contains
+codecs allowing the serialization of three classes:
+
+.. list-table:: Default Serializable Classes
+   :widths: 40 60
+   :stub-columns: 1
+   :header-rows: 1
+
+   * - Class
+     - Codec
+
+   * - :java-docs:`Document <apidocs/bson/org/bson/Document.html>`
+     - :java-docs:`DocumentCodec <apidocs/bson/org/bson/codecs/DocumentCodec.html>`
+
+   * - :java-docs:`BasicDBObject <apidocs/mongodb-driver-core/com/mongodb/BasicDBObject.html>`
+     - :java-docs:`DBObjectCodec <apidocs/mongodb-driver-core/com/mongodb/DBObjectCodec.html>`
+
+   * - :java-docs:`BsonDocument <apidocs/bson/org/bson/BsonDocument.html>`
+     - :java-docs:`BsonDocumentCodec <apidocs/bson/org/bson/codecs/BsonDocumentCodec.html>`
+
+Because the default ``CodecRegistry`` contains the codecs for these
+classes, you can use the ``Document``, ``BasicDBObject``, and
+``BsonDocument`` classes to represent data in a ``MongoCollection``
+without explicitly specifying a codec. For instance, the following
+example creates three instances of the ``exampleCollection`` collection
+in the ``exampleDatabase`` database, one for each class with
+serialization support in the default ``CodecRegistry``:
+
+.. code-block:: java
+
+   MongoClient client = MongoClients.create("mongodb+srv://<user>:<password>@<cluster-url>");
+   MongoDatabase database = client.getDatabase("exampleDatabase");
+   MongoCollection<Document> documentCollection =
+       database.getCollection("exampleCollection", Document.class);
+   MongoCollection<BasicDBObject> basicDBObjectCollection =
+       database.getCollection("exampleCollection", BasicDBObject.class);
+   MongoCollection<BsonDocument> BsonDocumentCollection =
+       database.getCollection("exampleCollection", BsonDocument.class);
+
+Custom Codecs
+-------------
+
+If the default codecs do not support pbject serialization for a class you need,
+you can specify a **custom codec**. Custom codecs specify how the driver
+should serialize data into :ref:`non-default classes <default-codecs>`.
+To serialize an unsupported class, you can:
```

Comment:
Nitpick/question: couldn't you use a custom codec to just change the way you want to de/serialize data between the EJSON and a class?

Reply:
Added a note about this alternative (but no less important) use case.